pallets/thea-executor/src/lib.rs

```diff
@@ -520,9 +562,8 @@ pub mod pallet {
 			let deposits: Vec<Deposit<T::AccountId>> =
 				Decode::decode(&mut &payload[..]).map_err(|_| Error::<T>::FailedToDecode)?;
 			for deposit in deposits {
-				<ApprovedDeposits<T>>::mutate(&deposit.recipient, |pending_deposits| {
-					pending_deposits.push(deposit.clone())
-				});
+				// Execute Deposit
+				Self::execute_deposit(deposit.clone())?;
 				Self::deposit_event(Event::<T>::DepositApproved(
```

Comment:
ApprovedEvent is getting emitted after the claim event. This is breaking the indexer as approved event happened before claim before.